pkg/cmd/release/update/update.go

```diff
@@ -0,0 +1,186 @@
+package update
+
+import (
+	"errors"
+	"fmt"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/spf13/cobra"
+	"net/http"
+)
+
+type UpdateOptions struct {
+	IO         *iostreams.IOStreams
+	HttpClient func() (*http.Client, error)
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	ReleaseId    string
+	TagName      string
+	Target       string
+	Name         string
+	Body         string
+	BodyProvided bool
+	Draft        *bool
+	Prerelease   *bool
+
+	// the value from the --repo flag
+	RepoOverride string
+}
+
+func NewCmdUpdate(f *cmdutil.Factory, runF func(*UpdateOptions) error) *cobra.Command {
+	opts := &UpdateOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+	}
+
+	var notesFile string
+
+	cmd := &cobra.Command{
+		DisableFlagsInUseLine: true,
+
+		Use:   "update <release_id>",
```

Comment:
gh users will likely not know the numeric release ID, so that shouldn't be part of the user interface. Instead, we let users select a release by tag name: see `gh help release view`.

https://github.com/cli/cli/blob/6783509d3bfdf55e1007dbedf34f2bdcd9fd1ec1/pkg/cmd/release/view/view.go#L92-L92

One limitation is that tag names cannot uniquely identify draft releases: there can exist a draft release without a tag name, or there can be multiple drafts for the same tag name. However, you need not solve that in your PR; we could discuss solutions in a separate issue, since a solution would apply to more than just this command.